src/classes/WalletManager/index.ts

```diff
@@ -76,20 +89,25 @@ export class WalletManager {
      * @returns {Promise<void>}
      */
     public async connect(
-        walletApplication: WalletApplication,
+        walletConnectorId: string,
         walletErrorHandlers?: ConnectWalletErrorHandlers,
     ): Promise<void> {
-        const provider = await detectEthereumProvider<MetaMaskEthereumProvider>()
+        const walletConnector = this._connectors[walletConnectorId]
+
+        if (!walletConnector)
+            throw new Error(EthError.PROVIDER_UNAVAILABLE)
+
+        const provider = await walletConnector.getProvider()
 
         if (!provider) {
             if (walletErrorHandlers?.onProviderUnavailable)
-                return walletErrorHandlers.onProviderUnavailable(walletApplication)
+                return walletErrorHandlers.onProviderUnavailable(walletConnector.id)
 
             throw new Error(EthError.PROVIDER_UNAVAILABLE)
         }
 
         try {
-            this._currentWalletApplication = walletApplication
+            this._currentWalletConnectorId = walletConnector.id
             this._nativeProvider = provider
             this._wrappedProvider = new BrowserProvider(this._nativeProvider as Eip1193Provider, 'any')
```

Comment:
i don't think the `as` will be required since connector now return a typed provider